pkg/streamingpromql/optimize/plan/commonsubexpressionelimination/optimization_pass.go

```diff
@@ -263,6 +268,7 @@ func (e *OptimizationPass) introduceDuplicateNode(group []*path, duplicatePathLe
 	duplicatedExpression, _ := firstPath.NodeAtOffsetFromLeaf(duplicatePathLength - 1)
 	duplicate := &Duplicate{Inner: duplicatedExpression, DuplicateDetails: &DuplicateDetails{}}
 	e.duplicationNodesIntroduced.Inc()
+	spanlogger.FromContext(ctx, log.NewNopLogger()).DebugLog("msg", "eliminated common subexpression", "expression_root", duplicatedExpression.Describe())
```

Comment:
I'm not sure this is going to be particularly helpful - `Describe()` will return  `"sum"` for an aggregation like `sum(foo)`.

Rather than adding this log line like this, what if we rely on the debugging line added above to know if CSE did anything, and rely on the analysis endpoint and UI for delving deeper into what happened?

Reply:
there's an analysis UI? 👀